man/loader.conf.xml

```diff
@@ -244,20 +244,29 @@
         <ulink url="https://edk2-docs.gitbook.io/understanding-the-uefi-secure-boot-chain/secure_boot_chain_in_uefi/uefi_secure_boot">
         EDK2 documentation</ulink>.</para>
 
-        <para>A complete set of UEFI variable includes <filename>db.esl</filename>, <filename>KEK.esl</filename>
-        and <filename>PK.esl</filename>. Note that these files need to be authenticated UEFI variables. See
+        <para>A complete set of UEFI variable includes <filename>db.auth</filename>, <filename>KEK.auth</filename>
+        and <filename>PK.auth</filename>. Note that these files need to be authenticated UEFI variables. See
         below for an example of how to generate them from regular X.509 keys.</para>
 
         <programlisting>uuid=$(systemd-id128 new --uuid)
 for key in PK KEK db; do
   openssl req -new -x509 -subj "/CN=${key}/" -keyout "${key}.key" -out "${key}.crt"
   openssl x509 -outform DER -in "${key}.crt" -out "${key}.cer"
-  cert-to-efi-sig-list -g "${uuid}" "${key}.crt" "${key}.tmp"
+  cert-to-efi-sig-list -g "${uuid}" "${key}.crt" "${key}.esl"
 done
 
-sign-efi-sig-list -c PK.crt -k PK.key PK PK.tmp PK.esl
-sign-efi-sig-list -c PK.crt -k PK.key KEK KEK.tmp KEK.esl
-sign-efi-sig-list -c KEK.crt -k KEK.key db db.tmp db.esl
+# This step adds the Microsoft Windows and UEFI keys to the database and is optional, but recommended.
+# The first key is needed to be able to use the Windows boot loader, the second key is needed for
+# firmware drivers and third-party boot loaders signed by Microsoft.
```

Comment:
drivers of the platform firmware don't have to be signed btw, they are trusted because loaded from a trusted storage source. 

I think we should mention the expression "option ROM" here, because that's what it really is about, i.e. stuff that is neither from the platform vendor nor from the OS vendor, but on some extension card.

Reply:
Agree, with @poettering here.

Should we maybe add an '(e.g. shim)' after third-party bootloader here? This would probably be the most common case.

Reply:
By the way, I think there has been multiple vulnerabilities discovered in signed version of the shim. Doesn't loading the microsoft key without loading any dbx for blacklisting these versions open you up to anyone booting the vulnerable shim and using it to do anything?

Maybe I'm thinking too far here.

Reply:
@poettering I would not trust a vendor to always have their internal drivers be trusted. I did add option ROM (it's really a terrible name, considering they are less than optional for nvidia GPUs, even on laptops).

@vdagonneau-anssi That's a good point. The nice thing is that we have nothing to do here: :D
Microsoft provides a third key (KEK) which is used to sign the official dbx updates. So I just added that key to the instructioins and referred to fwupdmgr to update the dbx.